.github/workflows/cd.yml

```
@@ -0,0 +1,171 @@
+      - name: Deploy to staging
+        run: |
+          echo "🚀 Deploying to staging environment"
+          echo "Image: ${{ needs.build-and-push.outputs.image-url }}"
+          # Add your staging deployment logic here
+          # For example: kubectl, Docker Swarm, or cloud provider specific commands
+
+  # Deploy to production
+  deploy-production:
+    name: Deploy to Production
+    runs-on: ubuntu-latest
+    needs: 
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)